### PR TITLE
[MTDSA-5908] Retrieve Other Expenses Controller, Route and ITs

### PR DESCRIPTION
**Pull request by Mbrookesy**

### CODE REVIEW

File: it/v1/endpoints/RetrieveOtherExpensesControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        response.json shouldBe responseBody
+        response.header("X-CorrelationId").nonEmpty shouldBe true
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
+    }
+
+    "return error according to spec" when {
+
+      "validation error" when {
+        def validationErrorTest(requestNino: String, requestId: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
```
Comment (Thepworth95) [inline it/v1/endpoints/RetrieveOtherExpensesControllerISpec.scala:125]: request id should be tax year

File: it/v1/endpoints/RetrieveOtherExpensesControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          s"validation fails with ${expectedBody.code} error" in new Test {
+
+            override val nino: String = requestNino
+            override val taxYear: String = requestId
```
Comment (Thepworth95) [inline it/v1/endpoints/RetrieveOtherExpensesControllerISpec.scala:129]: request id should be tax year

File: test/v1/controllers/RetrieveOtherExpensesControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(mtdError)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (TaxYearFormatError, BAD_REQUEST),
+          (RuleTaxYearRangeInvalidError, BAD_REQUEST),
```
Comment (Thepworth95) [inline test/v1/controllers/RetrieveOtherExpensesControllerSpec.scala:146]: don't need this error from the service

Comment (Mbrookesy) [reply test/v1/controllers/RetrieveOtherExpensesControllerSpec.scala:146]: Those set of errors aren't des errors, those are backend errors so RuleTaxYearRangeInvalid would be needed here

**Pull request merged**